src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -0,0 +1,255 @@
+package picard.util;
+
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.SequenceUtil;
+import htsjdk.variant.variantcontext.*;
+import picard.vcf.LiftoverVcf;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class LiftoverUtils {
+    /**
+     * This will take an input VariantContext and lift to the provided interval.
+     * If this interval is in the opposite orientation, all alleles and genotypes will be reverse complemented
+     * and indels will be left-aligned.  Currently this is only able to invert biallelic indels, and null will be
+     * returned for any unsupported VC.
+     * @param source The VariantContext to lift
+     * @param target The target interval
+     * @param refSeq The reference sequence, which should match the target interval
+     * @param writeOriginalPosition If true, INFO field annotations will be added to store the original position and contig
+     * @return The lifted VariantContext.  This will be null if the input VariantContext could not be lifted.
+     */
+    public static VariantContext liftVariant(VariantContext source, Interval target, ReferenceSequence refSeq, boolean writeOriginalPosition){
+        if (target == null) {
+            return null;
+        }
+
+        final VariantContextBuilder builder;
+        if (target.isNegativeStrand()){
+            builder = reverseComplementVariantContext(source, target, refSeq);
+        }
+        else {
+            builder = liftSimpleVariantContext(source, target);
+        }
+
+        if (builder == null){
+            return null;
+        }
+
+        builder.filters(source.getFilters());
+        builder.log10PError(source.getLog10PError());
+        builder.attributes(source.getAttributes());
+        builder.id(source.getID());
+
+        if (writeOriginalPosition) {
+            builder.attribute(LiftoverVcf.ORIGINAL_CONTIG, source.getContig());
+            builder.attribute(LiftoverVcf.ORIGINAL_START, source.getStart());
+        }
+
+        return builder.make();
+    }
+
+    protected static VariantContextBuilder liftSimpleVariantContext(VariantContext source, Interval target){
+        if (target == null || source.getReference().length() != target.length()) {
+            return null;
+        }
+
+        // Build the new variant context
+        final VariantContextBuilder builder = new VariantContextBuilder(source);
```

Comment:
Comment that this will copy over the genotypes and all the annotations, and filters

Reply:
ok